ed_sensor_integration/tools/fitter_viz.cpp

```diff
@@ -0,0 +1,382 @@
+//    {
+//        std::cout << "Error while reading file '" << filename << "':\n\n" << r.error() << std::endl;
+//        return false;
+//    }
+
+    return true;
+}
+
+// ----------------------------------------------------------------------------------------------------
+
+bool loadWorldModel(const std::string& model_name, ed::WorldModel& world_model)
+{
+    ed::UpdateRequest req;
+
+    ed::models::ModelLoader model_loader;
+
+    std::stringstream error;
+    if (!model_loader.create("_root", model_name, req, error, true))
+    {
+        std::cerr << "Model '" << model_name << "' could not be loaded:" << std::endl << std::endl;
+        std::cerr << error.str() << std::endl;
+        return false;
+    }
+
+    // Reset world
+    world_model = ed::WorldModel();
+
+    // Update world
+    world_model.update(req);
+
+    return true;
+}
+
+// ----------------------------------------------------------------------------------------------------
+
+void usage()
+{
+    std::cout << "Usage: ed_segmenter IMAGE-FILE-OR-DIRECTORY  WORLDMODEL_NAME  ENTITY_ID" << std::endl;
+}
+
+// Getting roll, pitch and yaw from a quaternion,
+// copied from https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles
+
+void drawLine( cv::Mat& canvas, geo::Vec2 point1, geo::Vec2 point2, geo::Transform2 pose, float resolution, int origin_x, int origin_y, cv::Scalar color)
+{
+    // computing points relative to the pose (in meters)
+    geo::Vec2 rel_point1 = pose * point1;
+    geo::Vec2 rel_point2 = pose * point2;
+
+    // position to pixels
+    int x_p1 = origin_x + (int)(rel_point1.x * resolution);
+    int y_p1 = origin_y - (int)(rel_point1.y * resolution);
+    int x_p2 = origin_x + (int)(rel_point2.x * resolution);
+    int y_p2 = origin_y - (int)(rel_point2.y * resolution);
+
+    if (x_p1 < 0 || x_p2 < 0|| x_p1 >= canvas.cols || x_p2 >= canvas.cols)
+    {
+        std::cout << "Entity: x-coordinate out of range" << std::endl;
+        return;
+    }
+    if (y_p1 < 0 || y_p2 < 0|| y_p1 >= canvas.rows || y_p2 >= canvas.rows)
+    {
+        std::cout << "Entity: y-coordinate out of range" << std::endl;
+        return;
+    }
+
+    // paint to screen
+    cv::Point point1_p(x_p1, y_p1);
+    cv::Point point2_p(x_p2, y_p2);
+    cv::line(canvas, point1_p, point2_p, color, 1);
+}
+
+void drawShape2D(cv::Mat& canvas, const Shape2D& shape, geo::Transform2 pose, float resolution, int origin_x, int origin_y, cv::Scalar color)
+{
+    for (unsigned int i=0; i < shape.size(); i++)
+    {
+        for (unsigned int j=0; j < shape[i].size()-1; j++)
+        {
+            drawLine(canvas, shape[i][j], shape[i][j+1], pose, resolution, origin_x, origin_y, color);
+        }
+        drawLine(canvas, shape[i][0], shape[i][shape[0].size()-1], pose, resolution, origin_x, origin_y, color);
+    }
+
+    // paint entity center
+
+    int x_p_ent = origin_x + (int)(pose.t.x * resolution);
+    int y_p_ent = origin_y - (int)(pose.t.y * resolution);
```

Comment:
Again, when using explicit casting, use modern style casting, `static_cast`/`dynamic_cast`, not the old c-style casting.

Again casting is equal to `floor`